src/main/java/sx/blah/discord/modules/ModuleLoader.java

```diff
@@ -208,26 +211,38 @@ public static synchronized void loadExternalModules(File file) { //A bit hacky,
 				Method method = URLClassLoader.class.getDeclaredMethod("addURL", new Class[]{URL.class});
 				method.setAccessible(true);
 				method.invoke(loader, new Object[]{url});
-
-				//Scans the jar file for classes which have IModule as a super class
-				List<String> classes = new ArrayList<>();
-				try (JarFile jar = new JarFile(file)) {
-					jar.stream().forEach(jarEntry -> {
-						if (!jarEntry.isDirectory() && jarEntry.getName().endsWith(".class")) {
-							String className = jarEntry.getName().replace('/', '.');
-							classes.add(className.substring(0, className.length()-".class".length()));
-						}
-					});
-				}
-				for (String clazz : classes) {
-					Class classInstance = Class.forName(clazz);
-					if (IModule.class.isAssignableFrom(classInstance)) {
-						addModuleClass(classInstance);
+				if (moduleClass == null) { //If the Module Developer has not specified the Implementing Class, revert to recursive search
+										//Scans the jar file for classes which have IModule as a super class
+					List<String> classes = new ArrayList<>();
+					try (JarFile jar = new JarFile(file)) {
+						jar.stream().forEach(jarEntry -> {
+							if (!jarEntry.isDirectory() && jarEntry.getName().endsWith(".class")) {
+								String className = jarEntry.getName().replace('/', '.');
+								classes.add(className.substring(0, className.length() - ".class".length()));
+							}
+						});
 					}
+					for (String clazz : classes) {
+						if (clazz.contains("$"))
+							continue;    //Avoids Nested Classes which can cause Errors due to instantiating before parent class
+						try {
+							Class classInstance = Class.forName(clazz);
+							if (IModule.class.isAssignableFrom(classInstance) && !classInstance.equals(IModule.class)) {
+								Discord4J.LOGGER.info(LogMarkers.MODULES, clazz);
+								addModuleClass(classInstance);
+							}
+						} catch (NoClassDefFoundError e) {	/*This can happen. Looking recursively looking through the classpath is hackish... */}
```

Comment:
Not sure I understand. I could move some brackets up lines so I have two brackets to a line but I think that would mess up the styling and make it more confusing